packages/compiler-cli/src/ngtsc/annotations/directive/src/initializer_functions.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import ts from 'typescript';
+
+import {ClassMember, ReflectionHost} from '../../../reflection';
+
+/**
+ * @fileoverview
+ *
+ * Angular exposes functions that can be used as class member initializers
+ * to make use of various APIs. Those are called initializer APIs.
+ *
+ * Signal-based inputs are relying on initializer APIs because such inputs
+ * are declared using `input` and `input.required` intersection functions.
+ * Similarly, signal-based queries follow the same pattern and are also
+ * declared through initializer APIs.
+ */
+
+export type InitializerApiFunction = 'input';
+
+/**
+ * Metadata describing an Angular class member that was recognized through
+ * a function initializer. Like `input`, `input.required` or `viewChild`.
+ */
+interface InitializerFunctionMetadata {
+  /** Name of the initializer API function that was recognized. */
+  apiName: InitializerApiFunction;
+  /** Node referring to the call expression. */
+  call: ts.CallExpression;
+  /** Whether the initializer is required or not. E.g. `input.required` was used. */
+  isRequired: boolean;
+}
+
+/**
+ * Attempts to identify an Angular class member that is declared via
+ * its initializer referring to a given initializer API function.
+ *
+ * Note that multiple possible initializer API function names can be specified,
+ * allowing for checking multiple types in one pass.
+ */
+export function tryParseInitializerApiMember<FnNames extends InitializerApiFunction[]>(
+    fnNames: FnNames, member: Pick<ClassMember, 'value'>, reflector: ReflectionHost,
+    isCore: boolean): InitializerFunctionMetadata&{apiName: FnNames[number]}|null {
+  if (member.value === null || !ts.isCallExpression(member.value)) {
+    return null;
+  }
+  const call = member.value;
+
+  // Extract target. Either:
+  //    - `[input]`
+  //    - `core.[input]`
+  //    - `input.[required]`
+  //    - `core.input.[required]`.
+  let target = extractPropertyTarget(call.expression);
+  if (target === null) {
+    return null;
+  }
+
+  // Find if the `target` matches one of the expected APIs we are looking for.
+  // e.g. `input`, or `viewChild`.
+  let apiName = fnNames.find(n => n === target?.text);
```

Comment:
```suggestion
  let apiName = fnNames.find(n => n === target!.text);
```

Given the non-null check this should be safe